### PR TITLE
fix: throw err when fail to get service name list

### PR DESCRIPTION
**Pull request by summer-ji-eng**
We should throw error if generated client library is empty.

Need feedback of error message @alexander-fenster @xiaozhenliu-gg5 🙏

### CODE REVIEW

File: typescript/test/unit/api.ts [hunk 1]
```diff
@@ -132,4 +140,16 @@ describe('schema/api.ts', () => {
       '../../protos/google/cloud/example/v1/test.proto',
     ]);
   });
+
+  it('should throw error when the service name is not found', () => {
+    assert.throws(() => {
```
Comment (alexander-fenster) [inline typescript/test/unit/api.ts]: Can you please move just the `new API()` line into `assert.throws`? This unit test checks that that exact constructor throws an exception, if anything else in this block throws for whatever reason, it should be a test failure.

```ts
it(....., { 
  stuff;
  stuff;
  stuff;
  assert.throws(() => { new API(...); });
});
```

File: typescript/src/schema/api.ts [hunk 2]
```diff
@@ -93,6 +93,11 @@ export class API {
         this.port = port ?? this.port ?? '443';
         serviceNamesList.push(service.name || this.naming.name);
       });
+    if (serviceNamesList.length === 0) {
+      throw new Error(
+        `Can't find ${this.naming.name}'s service names, please make sure that service names are defined in the proto file.`
```
Comment (alexander-fenster) [inline typescript/src/schema/api.ts]: The error message looks good! Let's maybe say `...make sure that services are defined...`, not the service names.

**Comment by xiaozhenliu-gg5** [comment]
Thanks Summer for doing this!
I have a question, since $version is also used to define the directory's name, should we check the version as well for the mainService? For example `monitoring` has an empty src folder except `index.ts` at the first time.

File: typescript/src/schema/api.ts (hunk 2 above)
Comment (summer-ji-eng) [reply typescript/src/schema/api.ts]: will do

File: typescript/test/unit/api.ts (hunk 1 above)
Comment (summer-ji-eng) [reply typescript/test/unit/api.ts]: agree

**Comment by summer-ji-eng** [comment]
> Thanks Summer for doing this!
> I have a question, since $version is also used to define the directory's name, should we check the version as well for the mainService? For example `monitoring` has an empty src folder except `index.ts` at the first time.

If I am understanding right, I believe that $version perform as expect. Before the fix, it generated correct folder struct, like:
> . test-out-monitoring
> > ...
>> src
>>> v3
>>>> index.js
>>> index.js
...
The content is empty in the index.js.

**Comment by alexander-fenster** [comment]
If there is no version, it will throw an exception in `naming.ts` being unable to parse the package name. I think we are good here!

**Pull request merged**